### DISCUSSION TITLE
Bluetooth Classic with Raspberry pi Pico

### DISCUSSION THREAD

**Discussion by jimseng** [micropython/micropython#12813] [General]
Hi All
The Raspberry pi Pico now supports Bluetooth BLE and classic according to the Raspberry Pi folks however the only examples available in micropython are for BLE. I have not found a mycropython example of, for instance, Bluetooth Classic SPP for the Rpi Pico and after asking on the Rpi Pico forum it was suggested I try here. I'm not sure if it is supported in micropython. Can someone suggest otherwise and help me out or is it still on the todo list?
Thanks

**Comment by jimmo** [comment]
There is not currently any support for BT Classic in MicroPython. This would be very nice to have, but not something that is currently planned.

It probably wouldn't be too difficult to supporting a small number of profiles (e.g. SPP/RFCOMM, HID, maybe A2DP). Mostly the challenge is API design :)

**Reply by zcattacz** [reply]
Profiles aside, what about HCI, is possible to use HCI somehow?

**Reply by jimmo** [reply]
This is not currently supported, the HCI link on the Pico W is a bit complicated (i.e. it's not a UART as is typical), but it would be possible to implement something that looked like a UART to expose to Python.

**Reply by jasmohan-narula** [reply]
@jimmo Any idea on how someone would go about implementing A2DP profile for ESP32S3? Any suggestions or references?

**Comment by mattytrentini** [comment]
If you are interested in implementing your own BT Classic library @jimseng, a good start would be to check out @shariltumin's BT Classic module for ESP32:

https://github.com/shariltumin/esp32-bluetooth-classic-micropython/

**Reply by jimseng** [reply]
I wish I had the skills!

**Reply by jimmo** [reply]
@mattytrentini This will help with what making Python bindings look like, but ESP32 uses a different host stack and there are some other considerations on the RP2040 with making it cooperate with the existing bluetooth module.

**Reply by mattytrentini** [reply]
Yeah, it was more an example of an API and how to implement bindings. The underlying Bluetooth code is indeed wildly different.
